zebra-rpc/src/methods/get_block_template_rpcs.rs

```diff
@@ -748,6 +758,47 @@ where
         }
         .boxed()
     }
+
+    fn validate_address(
+        &self,
+        raw_address: String,
+    ) -> BoxFuture<Result<validate_address::Response>> {
+        let network = self.network;
+
+        async move {
+            let Ok(address) = raw_address
+                .parse::<zcash_address::ZcashAddress>() else {
+                    return Ok(validate_address::Response::invalid());
+                };
+
+            let address = address
+                .convert::<primitives::Address>()
+                .map_err(|err| Error {
+                    code: ErrorCode::ServerError(0),
+                    message: format!("conversion error: {err}"),
+                    data: None,
+                })?;
+
+            return Ok(if address.network() == network {
+                validate_address::Response {
+                    address: Some(raw_address),
+                    is_valid: true,
+                    script_pub_key: None,
+                    pub_key: None,
+                    is_script: Some(address.is_script_hash()),
+                }
+            } else {
+                tracing::info!(
+                    ?network,
+                    address_network = ?address.network(),
+                    "network mismatch"
```

Comment:
Can we be more specific about which addresses don't match?
```suggestion
                    "invalid address in validateaddress RPC: Zebra's configured network and address network must match"
```

Reply:
Yep, updated in [`08ce203`](https://github.com/ZcashFoundation/zebra/pull/6086/commits/08ce20354dd0d9a51c4281c91b72c4617a362f4d).